### PR TITLE
Add Graphql client support

### PR DESCRIPTION
**Pull request by braineo**
Added support for autocomplete and syntax checking for graphql files as well as `gql` tag in `ts` or `js` files

### CODE REVIEW

**Review by ericdallo — COMMENTED** [review]
Cool! Could you add the proper documentation following https://emacs-lsp.github.io/lsp-mode/page/adding-new-language/#documentation please?

File: clients/lsp-graphql.el [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                        (or "ts" "js" "jsx" "tsx" "vue" "graphql" "gql")eos)
+        filename)
+    (and (derived-mode-p 'js-mode 'js2-mode 'typescript-mode)
+      (not (derived-mode-p 'json-mode)))))
+
+(lsp-register-client
+  (make-lsp-client :new-connection (lsp-stdio-connection (lambda()
+                                                           `(,(lsp-package-path 'graphql-language-service-cli)
+                                                              "server"
+                                                              "--method=stream")))
```
Comment (ericdallo) [inline clients/lsp-graphql.el:53]: Maybe we could extract the args to a defcustom for better customization?

Comment (braineo) [reply clients/lsp-graphql.el]: Sounds good. I took args out.

**Comment by braineo** [comment]
> Cool! Could you add the proper documentation following https://emacs-lsp.github.io/lsp-mode/page/adding-new-language/#documentation please?

I took a look at the `mkdocs.yml` and did not find anything about the actual `page/lsp-xxxx.md` files. Are they automatically generated?

**Comment by ericdallo** [comment]
Yes @braineo, the logic is on docs/lsp-doc.el, you can test the changes locally with `make local-webpage`

**Comment by braineo** [comment]
Very cool. I confirmed the doc working.

File: clients/lsp-graphql.el [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  :link '(url-link "https://github.com/graphql/graphiql/tree/main/packages/graphql-language-service-cli#readme")
+  :group 'lsp-mode)
+
+(defcustom lsp-clients-graphql-server-args '("server" "--method=stream")
+  "CLI arguments for graphql language server."
+  :type '(repeat string)
+  :risky t
+  :group 'lsp-graphql)
+
+(add-to-list 'lsp-language-id-configuration '(graphql-mode . "graphql"))
```
Comment (ericdallo) [inline clients/lsp-graphql.el:45]: All other clients are adding this direct to the variable `lsp-language-id-configuration`, this is the only one that is doing this different but I'm not sure this is a issue 🤔

Comment (braineo) [reply clients/lsp-graphql.el]: You are right. Better make it consistent. I started this client from my .emacs.d configure.

**Comment by braineo** [comment]
I don't understand how my changes caused `cask` to fail

**Comment by ericdallo** [comment]
That's infra issues @braineo not related to your PR

**Comment by ericdallo** [comment]
Thank you for your contribution!

**Pull request merged**